cli/azd/internal/appdetect/docker_test.go

```diff
@@ -0,0 +1,49 @@
+package appdetect
+
+import (
+	"github.com/stretchr/testify/assert"
+	"path/filepath"
+	"testing"
+)
+
+func TestParsePortsInLine(t *testing.T) {
+	tests := []struct {
+		portString    string
+		expectedPorts []Port
+	}{
+		{"", nil},
+		{"80", []Port{{80, "tcp"}}},
+		{"80 3100", []Port{{80, "tcp"}, {3100, "tcp"}}},
+		{"80 3100/udp", []Port{{80, "tcp"}, {3100, "udp"}}},
+		{" 80/tcp 3100/udp ", []Port{{80, "tcp"}, {3100, "udp"}}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.portString, func(t *testing.T) {
+			actual, err := parsePortsInLine(tt.portString)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedPorts, actual)
+		})
+	}
+}
+
+func TestParsePortsInFile(t *testing.T) {
+	tests := []struct {
+		dockerFileName string
+		expectedPorts  []Port
+	}{
+		{"DockerfileNoPort", nil},
```

Comment:
This looks good! One final comment here: instead of using files to capture the different test conditions, can we consider writing the EXPOSE lines here? 

```golang
tests := []struct {
		dockerFile string
		expectedPorts  []Port
	}{
		{"EXPOSE 80/tcp", []Port{{80, "tcp"}}},
		{"EXPOSE 80/tcp 3100/tcp", []Port{{80, "tcp"}, {3100, "tcp"}}},
		{"EXPOSE 80/tcp\n3100/tcp", []Port{{80, "tcp"}, {3100, "tcp"}}},
		{". EXPOSE 80/tcp\n3100/udp", []Port{{80, "tcp"}, {3100, "udp"}}},
}
```

Then we generate the file into a temp directory, and call `parsePortsInFile`. Keep in mind that these are actually perfectly valid (syntax-wise) format Dockerfiles, so there's some power in writing in this way. We could for example, at some point, decide to create variables to read from elsewhere, or just to keep it within the file -- any way works.

In the current implementation, I think we'd end up with *too many* files if we were to write test asset files for each test case. My proposal is trying to make it more concise to read and to maintain.

Reply:
Updated.